src/scripts/main.js

```diff
@@ -1,3 +1,39 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+
+function wrapHeadersWithSpan() {
+  const listItems = tree.querySelectorAll('li');
+
+  listItems.forEach((li) => {
+    if (li.querySelector('ul')) {
+      const textNode = li.childNodes[0];
```

Comment:
The code assumes that the first child node is always a text node. If the structure of the list items changes, this assumption might not hold true, leading to potential errors. Consider adding a check to ensure that `textNode` is indeed a text node before proceeding.